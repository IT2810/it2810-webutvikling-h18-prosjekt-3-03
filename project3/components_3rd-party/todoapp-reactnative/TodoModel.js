class TodoModel {
    constructor(title, completed, createdAt = new Date()) {
        this.title = title;
        this.completed = completed || false;
        this.createdAt = createdAt;
    }
}

module.exports = TodoModel;
