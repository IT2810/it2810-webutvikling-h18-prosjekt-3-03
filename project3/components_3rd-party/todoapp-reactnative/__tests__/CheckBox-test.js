import React from 'react';
import {View} from 'react-native';
import renderer from 'react-test-renderer';
import CheckBox from '../CheckBox';
import TodoModel from "../TodoModel";

it('renders correctly', () => {
    const tree = renderer
        .create(<View>
            <CheckBox data={new TodoModel("", false, new Date(0))}/>
            <CheckBox data={new TodoModel("", false, new Date(0))} color={'#000'}/>
            <CheckBox data={new TodoModel("", false, new Date(0))} color={'#C5C8C9'}/>
        </View>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
