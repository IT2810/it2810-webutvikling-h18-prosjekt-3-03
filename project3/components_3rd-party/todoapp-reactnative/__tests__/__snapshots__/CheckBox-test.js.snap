// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "backgroundColor": "rgba(0,0,0,0)",
          "borderRadius": 5,
        },
      ]
    }
  >
    <View
      data={
        TodoModel {
          "completed": false,
          "createdAt": 1970-01-01T00:00:00.000Z,
          "title": "",
        }
      }
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "padding": 8,
          },
          Object {
            "backgroundColor": "rgba(0,0,0,0)",
            "borderRadius": 5,
          },
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      />
    </View>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "backgroundColor": "rgba(0,0,0,0)",
          "borderRadius": 5,
        },
      ]
    }
  >
    <View
      data={
        TodoModel {
          "completed": false,
          "createdAt": 1970-01-01T00:00:00.000Z,
          "title": "",
        }
      }
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "padding": 8,
          },
          Object {
            "backgroundColor": "rgba(0,0,0,0)",
            "borderRadius": 5,
          },
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      />
    </View>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "backgroundColor": "rgba(0,0,0,0)",
          "borderRadius": 5,
        },
      ]
    }
  >
    <View
      data={
        TodoModel {
          "completed": false,
          "createdAt": 1970-01-01T00:00:00.000Z,
          "title": "",
        }
      }
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "padding": 8,
          },
          Object {
            "backgroundColor": "rgba(0,0,0,0)",
            "borderRadius": 5,
          },
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      />
    </View>
  </View>
</View>
`;
